### DISCUSSION TITLE
return run id while streaming output with RemoteRunnable

### DISCUSSION THREAD

**Discussion by justas145** [langchain-ai/langserve#700] [Q&A]
I have a remote runnable:
remoteChain = RemoteRunnable("http://localhost:8420/chat")

and I am streaming an output:

for chunk in remoteChain.stream({'question': "My question", 'user_id': user_id'}, config={"configurable": {
    "conversation_id": "conv_id",
    "user_id": "user_id"
}}):
    print(chunk)

I would like to get the run id as well from remoteChain. Is that possible?

**Comment by eyurtsev** [comment]
1. You can provide `run_id` from the client, as long as you configure server to accept it (not sure if it does by default) -- check add_routes documentation.
2. stream() already returns run_id, but RemoteRunnable doesn't have access to it right now, you could access it via a raw client, or else sub-class and override to define a callback for the run id
